File: src/main/java/org/minturtle/careersupport/codereview/respository/ReviewPinpointRepository.java
package org.minturtle.careersupport.codereview.respository;

import org.minturtle.careersupport.codereview.entity.CommitPinpoint;
import org.minturtle.careersupport.common.aop.Logging;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

@Logging
public interface ReviewPinpointRepository extends ReactiveMongoRepository<CommitPinpoint, String> {
    Mono<CommitPinpoint> findByPrNumber(Integer prNumber);
}

File: src/main/kotlin/org/minturtle/careersupport/codereview/respository/ReviewPinpointRepository.kt
package org.minturtle.careersupport.codereview.respository

import org.minturtle.careersupport.codereview.entity.CommitPinpoint
import org.springframework.data.repository.kotlin.CoroutineCrudRepository

interface ReviewPinpointRepository : CoroutineCrudRepository<CommitPinpoint, String> {

    suspend fun findByPrNumberAndRepositoryName(prNumber: Int, repositoryName: String): CommitPinpoint?

}